### ISSUE TITLE
`UseTransport` creates an instance of a `TransportDefinition` for every invocation

### ISSUE THREAD

**Issue by danielmarbach** [Particular/NServiceBus#3289]
Due to the nature of how we setup acceptance tests we need to call `UseTransport` potentially multiple times, which happens to throw away everything previously setup on the transport definition of the currently setup test. 

Since the common shared state is placed into the `SettingsHolder` the `UseTransport` extension method should only create the `TransportDefinition` when there is not already one available. If there is already one available the `UseTransport` call only create a new `TransportExtension` and return the extension.

In short

* `TransportDefinition` becomes stateful per `SettingsHolder` instance
* `TransportExtensions` remains stateless

This is currently blocking `AzureServiceBus` acceptance test implementations.

**Comment by johnsimons** [comment]
Doesn't the same issue exists for the other "Definitions" ?

**Comment by danielmarbach** [comment]
@johnsimons I think yes

**Comment by yvesgoeleven** [comment]
@johnsimons what others are there, I know about PersistenceDefinition, any others?

**Comment by johnsimons** [comment]
@yvesgoeleven I think there is also the serializers

**Comment by danielmarbach** [comment]
Situation improved with https://github.com/Particular/NServiceBus/commit/f25bc7e800bc468507350c8238a769ddc3aa83ef. Closing this
